### PR TITLE
5.2.0 refactoring

### PR DESCRIPTION
**Pull request by johannnallathamby**

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/pom.xml [hunk 1]
```diff
@@ -74,6 +74,14 @@
             <groupId>org.wso2.securevault</groupId>
             <artifactId>org.wso2.securevault</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.commons</groupId>
+            <artifactId>org.wso2.carbon.tenant.common</artifactId>
+        </dependency>
+        <dependency>
```
Comment (cdwijayarathna) [inline components/identity-core/org.wso2.carbon.identity.core/pom.xml]: This introduces cyclic dependency, user.store.configuration is depending on identity.core, we can't put identity.core to depend on it.

File: components/identity-core/org.wso2.carbon.identity.core/pom.xml [hunk 2]
```diff
@@ -74,6 +74,14 @@
...
+            <groupId>org.wso2.carbon.identity</groupId>
+            <artifactId>org.wso2.carbon.identity.user.store.configuration</artifactId>
+        </dependency>
```
Comment (cdwijayarathna) [inline components/identity-core/org.wso2.carbon.identity.core/pom.xml]: This introduces cyclic dependency, user.store.configuration is depending on identity.core, we can't put identity.core to depend on it.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/InboundConstants.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (madurangasiriwardena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/InboundConstants.java]: No need to change the licence year

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/InboundMessageContext.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+	protected InboundRequest request;
+    protected Map<T1,T2> parameters = new HashMap<>();
+
+    public InboundMessageContext(InboundRequest request, Map<T1, T2> parameters){
+        super(parameters);
+        this.request = request;
+    }
+
+	public InboundRequest getRequest() {
```
Comment (madurangasiriwardena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/InboundMessageContext.java]: formatting issue

**Comment by Pushpalanka** [comment]
Waiting for review suggestions to be addressed before merging....

**Pull request merged**